ir/instr.cpp

```diff
@@ -3159,15 +3159,17 @@ void Memset::print(ostream &os) const {
 }
 
 StateValue Memset::toSMT(State &s) const {
-  auto &vbytes = s.getAndAddPoisonUB(*bytes).value;
+  auto &vbytes = s.getAndAddPoisonUB(*bytes, true).value;
 
   uint64_t n;
   expr vptr;
   if (vbytes.isUInt(n) && n > 0) {
-    vptr = s.getAndAddPoisonUB(*ptr).value;
+    vptr = s.getAndAddPoisonUB(*ptr, true).value;
   } else {
     auto &sv_ptr = s[*ptr];
-    s.addUB((vbytes != 0).implies(sv_ptr.non_poison));
+    auto &sv_ptr2 = s[*ptr];
+    s.addUB((vbytes != 0).implies(
+        sv_ptr.non_poison && (sv_ptr.value == sv_ptr2.value)));
```

Comment:
This naively encodes non-undefness, but I think we need to encapsulate this in a new function in State or somewhere that returns (value, non-undefness).
I can work on this if you want.